chronicle/Utilities/ApiUtils.swift

```diff
@@ -0,0 +1,66 @@
+//
+//  ApiUtils.swift
+//  ApiUtils
+//
+//  Created by Alfonce Nzioka on 8/11/21.
+//
+
+import Foundation
+
+/// Utility functions for API calls
+struct ApiUtils {
+    
+    struct ChronicleApi {
+        
+        static let scheme = "https"
+        static var host = "api.openlattice.com"
+        static let v2Base = "/chronicle/v2"
+        static let base = "/chronicle"
+        
+        // path constants
+        static let enrollPath = "/enroll"
+        static let studyPath = "/study"
+    }
+    
+    // returns an optional URLComponent with orgId in the path
+    static func makeEnrollDeviceComponentsWithOrg (enrollment: Enrollment, deviceId: String) -> URLComponents? {
+        guard EnrollmentUtils.validateEnrollmentDetails(enrollment: enrollment, withOrgId: true) else {
+            return nil
+        }
+        
+        guard !deviceId.isEmpty else {
+            return nil
+        }
+        
+        var components = URLComponents()
+        
+        components.scheme = ChronicleApi.scheme
+        components.host = ChronicleApi.host
+        components.path = "\(ChronicleApi.v2Base)/\(enrollment.organizationId)/\(enrollment.studyId)/\(enrollment.participantId)/\(deviceId)\(ChronicleApi.enrollPath)"
```

Comment:
Are you manually constructing the url path? Am I reading this correctly? If yes, I think this is fine for now but we should revisit in the future.

Reply:
Yes. Probably inevitable with swift